src/liballoc/collections/btree/set.rs

```diff
@@ -1055,6 +1086,46 @@ impl<'a, T> IntoIterator for &'a BTreeSet<T> {
     }
 }
 
+/// An iterator produced by calling `drain_filter` on BTreeSet.
+#[unstable(feature = "btree_drain_filter", issue = "42849")]
+pub struct DrainFilter<'a, T>(
```

Comment:
This ought to have an `F` parameter for API stability, and everything should be constrained as if we're using `F` directly. This way it's _possible_ to remove the box later, even if we just add `PhantomData<F>` for now.